sdk/program/src/stake_history.rs

```diff
@@ -81,9 +87,83 @@ impl Deref for StakeHistory {
     }
 }
 
+// HANA my stuff below
+
+#[derive(Debug, PartialEq, Eq, Default, Clone)]
+pub struct StakeHistoryData<'a>(Rc<RefCell<&'a mut [u8]>>);
+
+// HANA should i impl all of Sysvar for this?
+impl<'a> StakeHistoryData<'a> {
+    pub fn from_account_info(account_info: &AccountInfo<'a>) -> Result<Self, ProgramError> {
+        if *account_info.unsigned_key() != StakeHistory::id() {
+            return Err(ProgramError::InvalidArgument);
+        }
+        Ok(StakeHistoryData(account_info.data.clone()))
+    }
+}
+
+// HANA i would like to change this to Option<&StakeHistoryEntry> but the types are complicated
+// i dont think its possible to take a reference to the data inside the RefCell
+// i also dont think theres any way to heap allocation in my new impl and get a normal reference
+// and i dont think theres a way to use something like AsRef to let me box my new one and not the old one
+// and anyway callers of the old one wouldnt have a type specialized to normal reference
+// so really i would need like... hmm i can put `ReturnType: AsRef<StakeHistoryEntry>`
+// and have each impl define its own return, Option<&StakeHistoryEntry> or Option<Box<StakeHistoryEntry>>
+pub trait StakeHistoryGetEntry {
+    fn get_entry(&self, epoch: Epoch) -> Option<StakeHistoryEntry>;
+}
```

Comment:
flagging for visibility. i dont know if theres hot code that really benefits from the reference return type, or if the copy is fine